xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/internal/html/DefaultHTMLCleanerTest.java

```diff
@@ -104,7 +104,9 @@ void specialCharacters()
         assertHTML("<p>\"&amp;</p>", "<p>\"&</p>");
         assertHTML("<p><img src=\"http://host.com/a.gif?a=foo&amp;b=bar\" /></p>",
             "<img src=\"http://host.com/a.gif?a=foo&amp;b=bar\" />");
-        assertHTML("<p>&#xA;</p>", "<p>&#xA;</p>");
+        assertHTML("<p>\n</p>", "<p>&#xA;</p>");
```

Comment:
this behavior is new in the 2.25 version: line breaks are now inserted verbatim instead of encoded.

Reply:
I'm not sure we want this behavior to change.

Reply:
I guess I understand why you do not want to change this, but this is hard to achieve, if a literal `\n` line break should also be preserved in the output (as it is tested in `doNotRemoveLFandCRInTextNodes`) and implicitly in various other places.
The problem is that the input string is first converted into a DOM, and for text nodes in the DOM there is only the character `\n` to represent both  `&#xA;` and `\n` (and also `&#10;`, etc). The information if the character has been represented by its unicode entity representation or its raw value has been lost at this point.

When the `setRecognizeUnicodeChars` is set to false, however, any `&#xA;` is replaced by `&amp;#xA;` (as is the same for all other character entities), but then that is no longer distinguishable from the user input `&amp;#xA;`, which is the problem of XCOMMONS-2265.  The only way I see around this is either to replace `\n' or `&#xA;` by some marker string not present in the input string and then replace that marker string in the output by the character it represents. However that seems a very ugly hack for me.

Reply:
Well we actually do perform automatic transformation of entities, the first assert of this test shows that we don't keep `&nbsp;` and `&quot;` as they are for example. So maybe unicode characters should be transformed to, note that we currently use in our default properties: `defaultProperties.setDeserializeEntities(true);` so this behaviour actually makes sense. 
I guess we need to ensure it won't cause any trouble for the users. I'm not entirely sure on my side, maybe @mflorea would have some feedback on that?

Reply:
I don't see any reason to keep escaped characters that don't actually need escaping. We don't really lose any information.